src/avram/upsert.cr

```diff
@@ -90,6 +90,24 @@ module Avram::Upsert
     end
   end
 
+  macro upsert_unique_on(*attribute_names)
+    def self.bulk_upsert(upserts)
+      operations = upserts.map do |upsert_args|
+        new(**upsert_args)
```

Comment:
As of now, when a user passes a mismatching array, the compiler throws an error that double splats don't support unions
It would be _awesome_ if we could catch this and provide a clean message that all records must have matching keys

Reply:
The error when it doesn't compile should be something like:
> You must pass matching NamedTuples into upsert

rather than complaining about a `**`, which isn't going to be clear to newcomers

Reply:
https://play.crystal-lang.org/#/r/cmgb is a viable approach here